feat: Sanitize iframe URL

yagopv opened the pull request:
## What it solves
Sanitize iframe URL for avoid any JS execution

usame-algan commented on src/routes/safe/components/Apps/utils.ts:
```diff
@@ -200,3 +200,48 @@ export const getLegacyChainName = (chainName: string, chainId: string): string =
+  })
+}
+
+export function sanitizeUrl(url: string | null): string {
```
I would suggest adding a few unit-tests for positive and negative cases.

yagopv replied:
Added

iamacook reviewed (commented):
The `isSameURL()` check will fail [here](https://github.com/safe-global/safe-react/blob/c059d9baa042f0c1f14fe21bec1dc678f6d3ce8f/src/routes/safe/components/Apps/components/AppFrame.tsx#L161) as your comparing the sanitized URL with the original `appUrl`. It's probably best to santiize inside `useSafeAppUrl()`. What do you think?

katspaugh commented:
I don't think you need HTML entity unescaping. React isn't inserting the iframe URL in HTML, it sets it as a DOM property.
So if someone passes a URL in HTML encoding, it will be inserted as is.

yagopv commented:
> I don't think you need HTML entity unescaping. React isn't inserting the iframe URL in HTML, it sets it as a DOM property. So if someone passes a URL in HTML encoding, it will be inserted as is.

Yeah I can remove it. I took the script from a library and is working with more cases than our concrete one

katspaugh commented:
It would be nice to show an error straight away.

<img width="1513" alt="Screenshot 2022-04-25 at 14 27 29" src="https://user-images.githubusercontent.com/381895/165088808-83602a3d-fbcf-4862-ad52-8ca6d969b2bb.png">

After N seconds:
<img width="1469" alt="Screenshot 2022-04-25 at 14 27 45" src="https://user-images.githubusercontent.com/381895/165088906-a25ffea4-19a8-47c3-ba0d-630259901dac.png">

yagopv commented:
> The `isSameURL()` check will fail [here](https://github.com/safe-global/safe-react/blob/c059d9baa042f0c1f14fe21bec1dc678f6d3ce8f/src/routes/safe/components/Apps/components/AppFrame.tsx#L161) as your comparing the sanitized URL with the original `appUrl`. It's probably best to santiize inside `useSafeAppUrl()`. What do you think?

I did it. Moved to the hook. I changed the location of the function itself and added the tests

katspaugh commented:
Redirecting to Apps IMHO would be no bueno, as it will not alert the user and hide the problem.
If they stay on that page and see an error, they will be aware of potential phishing attempts.

yagopv commented:
> Redirecting to Apps IMHO would be no bueno, as it will not alert the user and hide the problem.
> If they stay on that page and see an error, they will be aware of potential phishing attempts.

I like the error approach. Take a look now. I'm throwing an error if the sanitize call detect a `javascript:` injection warning the user

katspaugh commented:
It crashes the app immediately now, good 👍 

The error is generic tho, something we could improve. But good enough for now.
<img width="475" alt="Screenshot 2022-04-25 at 16 35 49" src="https://user-images.githubusercontent.com/381895/165111419-8b595003-452b-435b-98b8-39ac5c9d603f.png">

iamacook reviewed (approved):
Nice work 🚀

yagopv commented:
@JagoFigueroa did you test this?

JagoFigueroa commented:
Excelente implementación camarada, looks good to me as well.

Un saludo ;)
